Backpack-SwiftUI/Select/Classes/BPKSelect.swift

```diff
@@ -0,0 +1,170 @@
+                    BPKText(
+                        labelText
+                    )
+                    .foregroundColor(textColor)
+                    Spacer()
+                    Image(systemName: "triangle.fill")
+                        .resizable()
+                        .foregroundColor(.textPrimaryColor)
+                        .frame(width: 8, height: 5)
+                        .rotationEffect(.degrees(180))
+                }
+            }
+            .frame(height: .lg)
+        }
+        .padding(.md)
+    }
+}
+
+extension View {
+    func customPickerStyle(labelText: String,
+                           textColor: BPKColor) -> some View {
+        self.modifier(
+            CustomPickerStyle(
+                labelText: labelText,
+                textColor: textColor
+            )
+        )
+    }
+}
+
+public struct BPKSelect: View {
+    
+    @SwiftUI.State private var selectedIndex: Int
+
+    private let options: [String]
+    private let title: String
+    private var state: State = .default
+    private let onSelectionChange: (Int) -> Void
+    
+    private var indexIdentifiableOptions: [IdentifiableString] {
+        options.enumerated().map { (index, element) in
+            IdentifiableString(string: element, id: index)
+        }
+    }
+    
+    public init(
+        placeholder: String,
+        options: [String],
+        selectedIndex: Int,
+        onSelectionChange: @escaping (Int) -> Void
+    ) {
+        self.options = options
+        self.selectedIndex = selectedIndex
+        self.title = placeholder
+        self.onSelectionChange = onSelectionChange
+    }
+    
+    public var body: some View {
+        VStack {
+            
+            Picker(title, selection: $selectedIndex) {
+                ForEach(indexIdentifiableOptions) { t in
+                    Text(t.string)
+                }
+            }
+            .customPickerStyle(
+                labelText: selectedIndex < options.count ? options[selectedIndex] : "",
+                textColor: state.textColor
+            )
+            .background(.surfaceDefaultColor)
+            .clipShape(RoundedRectangle(cornerRadius: .sm))
+            .disabled(state.isDisabled)
+            .outline(state.borderColor, cornerRadius: .sm)
+            .onChange(of: selectedIndex) { tag in
+                onSelectionChange(tag)
+            }
+        }
+    }
+    
+    public func inputState(_ state: State) -> BPKSelect {
+        var result = self
+        result.state = state
+        return result
+    }
+}
+
+struct BPKSelect_Previews: PreviewProvider {
+    static var previews: some View {
+        VStack {
+            BPKText("Breakfast Choices", style: .heading1)
+            BPKSelect(
+                placeholder: "Breakfast Choices",
+                options: ["Porridge",
+                          "Eggs",
+                          "Swift UI"],
+                selectedIndex: 99) { index in
```

Comment:
can you replace this with `{ _ in }`? same with the others!

Reply:
changed in: https://github.com/Skyscanner/backpack-ios/pull/1708/commits/e869ab618cf4f5b4bd63c6bcc874610a21b8d83a